nobrainer/processing/generation.py

```diff
@@ -0,0 +1,191 @@
+import importlib
+import os
+
+import tensorflow as tf
+
+from .base import BaseEstimator
+from .. import losses
+from ..dataset import get_dataset
+from ..training import ProgressiveGANTrainer
+
+
+class Generation(BaseEstimator):
+    """Perform generation type operations"""
+
+    # state_variables = ["block_shape_", "volume_shape_", "scalar_labels_"]
+    state_variables = ["resolution_batch_size_map_"]
+
+    def __init__(self, base_model, model_args=None):
+        if not isinstance(base_model, str):
+            self.base_model = base_model.__name__
+        else:
+            self.base_model = base_model
+        self.model_ = None
+        self.model_args = model_args or {}
+        self.resolution_batch_size_map_ = None
+        self.reolutions_ = sorted(list(self.resolution_batch_size_map_.keys()))
+        self.file_pattern = "*res-%03d*.tfrec"
+
+    def fit(
+        self,
+        dataset_train,
+        dataset_validate=None,
+        epochs=2,
+        checkpoint_dir=os.getcwd(),
+        latent_size=256,
+        g_fmap_base=1024,
+        d_fmap_base=1024,
+        num_channels=1,
+        dimensionality=3,
+        # TODO: figure out whether optimizer args should be flattened
+        g_optimizer=None,
+        g_opt_args=None,
+        d_optimizer=None,
+        d_opt_args=None,
+        g_loss=losses.wasserstein,
+        d_loss=losses.wasserstein,
```

Comment:
aren't these all model args?

Reply:
yes, should I use `self.model_args `  for these? I thought they should have a default value somewhere.
or I can do like `opt_args`?